pkg/code-generator/codegen/templates/event_loop_template.go

```diff
@@ -67,7 +67,7 @@ func (el *{{ lower_camel .GoName }}EventLoop) Run(namespaces []string, opts clie
 	go func() {
 		// create a new context for each loop, cancel it before each loop
 		var cancel context.CancelFunc = func() {}
-		defer func() { cancel() }()
+		defer cancel()
```

Comment:
the reason i used the more complicated syntax is because cancel is a variable,
and i want to make sure that defer uses it's latest value, and not its current value (as its value is changed in the select block.

I'm not 100% sure of the defer semantics, and there's a chance that this will work, I'm ok with this change if you can add a unit tests that proves that this works